broker/src/main/java/io/camunda/zeebe/broker/partitioning/topology/TopologyPartitionListenerImpl.java

```diff
@@ -7,24 +7,25 @@
  */
 package io.camunda.zeebe.broker.partitioning.topology;
 
+import io.camunda.zeebe.engine.api.ProcessingScheduleService;
 import io.camunda.zeebe.protocol.impl.encoding.BrokerInfo;
 import io.camunda.zeebe.protocol.record.PartitionRole;
-import io.camunda.zeebe.scheduler.ActorControl;
+import java.time.Duration;
 import org.agrona.collections.Int2IntHashMap;
 
 public final class TopologyPartitionListenerImpl implements TopologyPartitionListener {
 
   private final Int2IntHashMap partitionLeaders = new Int2IntHashMap(-1);
-  private final ActorControl actor;
+  private final ProcessingScheduleService scheduleService;
 
-  public TopologyPartitionListenerImpl(final ActorControl actor) {
-    this.actor = actor;
+  public TopologyPartitionListenerImpl(final ProcessingScheduleService scheduleService) {
+    this.scheduleService = scheduleService;
   }
 
   @Override
   public void onPartitionLeaderUpdated(final int partitionId, final BrokerInfo member) {
     if (member.getPartitionRoles().get(partitionId) == PartitionRole.LEADER) {
-      actor.submit(() -> updatePartitionLeader(partitionId, member));
+      scheduleService.runDelayed(Duration.ZERO, () -> updatePartitionLeader(partitionId, member));
```

Comment:
❓ Does it make sense to run something with a delay of 0 seconds?

Reply:
Looks fine to me, just means run this as soon as you can. That said, if we see it's a repeated pattern, we could think of adding some `runAsync` to differentiate usages and properly implement it on our side.